File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0) return false;
        
        int reversedNum = 0;
        int xCopy = x;
        while(xCopy>0){
            reversedNum *= 10;
            reversedNum += xCopy%10;
            
            xCopy = xCopy/10;
        }
        // System.out.println(xCopy);
        // System.out.println(reversedNum);
        
        if(reversedNum == x){
            return true;
        }
        return false;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
	def isPalindrome(self, n):
	    bool = True 
	    n = str(n)
	    for i in range(len(n)):
	        if len(n)-i-1 >= i and n[i] != n[len(n) -i -1]:
	            bool = False
	            break
	    return bool